src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage_WriteBatching.cs

```diff
@@ -129,8 +129,13 @@ async Task<(Task previousTask, TaskCompletionSource<int> taskCompletionSource)>
                 // performed by FlushAllPendingWritesAsync.
                 using (await _writeQueueGate.DisposableWaitAsync(cancellationToken).ConfigureAwait(false))
                 {
-                    // Get the writes we need to process.
-                    writesToProcess.AddRange(keyToWriteActions[key]);
+                    // Get the writes we need to process. 
+                    // Note: explicitly foreach so we operate on the struct enumerator for
+                    // MultiDictionary.ValueSet.
+                    foreach (var action in keyToWriteActions[key])
```

Comment:
Can we extend the capacity with `keyToWriteActions.Count` first?

Reply:
Let me check!

Reply:
Done.